MalomAPI_cpp/perfect_common.h

```diff
@@ -0,0 +1,346 @@
+/*
+Malom, a Nine Men's Morris (and variants) player and solver program.
+Copyright(C) 2007-2016  Gabor E. Gevay, Gabor Danner
+Copyright (C) 2023 The Sanmill developers (see AUTHORS file)
+
+See our webpage (and the paper linked from there):
+http://compalg.inf.elte.hu/~ggevay/mills/index.php
+
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef PERFECT_COMMON_H_INCLUDED
+#define PERFECT_COMMON_H_INCLUDED
```

Comment:
(Are you using these instead of the `#pragma once` that is in the original code, because you also compile this project with something other than Visual C++?)

Reply:
Thank you for your insightful comment regarding the use of include guards over `#pragma once`.

You are correct in noting that the original code utilizes `#pragma once`. However, I opted for traditional include guards for the sake of broader compiler compatibility, including GCC and others, as well as cross-platform operability.

1. **Compiler Compatibility**: While `#pragma once` is widely supported, it is not part of the C++ standard. Traditional include guards are standard-compliant and thus ensure compatibility with a broader set of compilers.

2. **Cross-Platform Support**: This software "Sanmill" needs to run on multiple platforms including Linux and macOS. The traditional include guard approach is more portable and guarantees that the code will behave consistently across different operating systems and file systems.

3. **Debian Inclusion**: One of the long-term objectives for Sanmill project is to have it officially included in Debian distributions. Adhering to best practices for maximum compatibility aligns well with this goal.

Therefore, the decision to use traditional include guards was made with these considerations in mind, aiming to make the code as robust and portable as possible.

Thank you again for your thoughtful question, and I appreciate any further comments you may have.